modules/Components/src/client/ACASThingBrowser.coffee

```diff
@@ -331,42 +380,83 @@ class ACASThingBrowserController extends Backbone.View
+		for c in cls
+			@$("." + c).addClass("hide")
+		if cl?
+			@$("." + cl).removeClass("hide")
+		@$(".bv_thingTableController").addClass("hide")
+		@$(".bv_searchThingsStatusIndicator").removeClass("hide")
+
+	moreSpecificSearchRequired: =>
+		@showOne("bv_moreSpecificThingSearchNeeded")
+
+	searchingStarted: =>
 		@destroyThingSummaryTable()
+		@showOne("bv_searchingThingsMessage")
 
-		@$(".bv_searchingThingsMessage").addClass "hide"
-		if things is null
-			@$(".bv_errorOccurredPerformingSearch").removeClass "hide"
+	setupThingSummaryTable: (response) =>
+		results = response.results
 
-		else if things.length is 0
-			@$(".bv_noMatchingThingsFoundMessage").removeClass "hide"
-			@$(".bv_thingTableController").html ""
+		if response.maxResults <= response.numberOfResults
+			msg = "Browser results were limited to the first " + response.maxResults + " entries out of " + response.numberOfResults + " results"
+			@$('.bv_maxThingBrowserSearchResultsReached').html msg
+			@$("bv_maxThingBrowserSearchResultsReached").removeClass("hide")
+		else 
+			@$("bv_maxSearchResultsLimited").addClass("hide")
+
+		if results is null
+			@showOne("bv_errorOccurredPerformingSearch")
+
+		else if results.length is 0
+			@showOne("bv_noMatchingThingsFoundMessage")
 		else
 			@$(".bv_searchThingsStatusIndicator").addClass "hide"
-			@$(".bv_thingTableController").removeClass "hide"
-			thingCollection =  Backbone.Collection.extend 
-				model: @modelClass
+			@$(".bv_thingTableController").removeClass "hide" 
+
 			@thingSummaryTable = new ThingSummaryTableController
-				collection: new thingCollection things
+				collection: results
+				thingModelClass: @modelClass
 				configs: @configs
 				columnFilters: @columnFilters
+				maxResults: @maxResults
 
 			@thingSummaryTable.on "selectedRowUpdated", @selectedThingUpdated
 			@$(".bv_thingTableController").html @thingSummaryTable.render().el
 
 	selectedThingUpdated: (thing) =>
 		@$('.bv_thingControllerWrapper').append("<div class='bv_thingController'></div>")
 		@trigger "selectedThingUpdated"
-		@thingController = new @controllerClass
-			el: @$('.bv_thingController')
-			model: thing
-			readOnly: true
 
+		# If thing is thing class then we pass it to the thing controller, if not then we fetch it from the server and pass it to the thing controller
+		if thing instanceof Thing
+			@thingController = new @controllerClass
+				model: thing
+				configs: @configs
+				el: @$('.bv_thingController')
+			@renderController()
+		else
+			c = new @modelClass(thing.attributes)
+			c.fetch
+				success: (model, response, options) =>
+					@thingController = new @controllerClass
+						model: model
+						configs: @configs
+						el: @$('.bv_thingController')
+					@renderController()
+				error: (model, response, options) =>
+					@$('.bv_thingController').html "Error fetching thing"
```

Comment:
When a row is clicked, if the row is not a Thing instance (it's a plain javascript object by default now), then we use the row (which includes the codeName and id) and fetch it from the server and pass it to the form editor controller.